src/Authenticator/JwtAuthenticator.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Authenticator;
+
+use Authentication\Identifier\IdentifierCollection;
+use Authentication\Result;
+use Cake\ORM\Entity;
```

Comment:
> According to the spec the subject in a token can be a JSON string.

@burzum  Which specs are you referring to? This is what the [JWT specs rfc7519](https://tools.ietf.org/html/rfc7519#section-4.1.2) says about "sub":

> The "sub" value is a case-sensitive string containing a StringOrURI  value.

So value a "sub" is always a string. So one can just set "id" or "username" value to sub and then use the `TokenAuthenticator` to return matching user record.